vision-quickstart/app/src/main/java/com/google/mlkit/vision/demo/VisionProcessorBase.java

```diff
@@ -153,7 +158,30 @@ private void processImage(
                 /* shouldShowFps= */ true)
                 .addOnSuccessListener(executor, results -> processLatestImage(graphicOverlay));
     }
+    @RequiresApi(api = VERSION_CODES.KITKAT)
+    private Bitmap toBitmap(Image image) {
+        Image.Plane[] planes = image.getPlanes();
+        ByteBuffer yBuffer = planes[0].getBuffer();
+        ByteBuffer uBuffer = planes[1].getBuffer();
+        ByteBuffer vBuffer = planes[2].getBuffer();
 
+        int ySize = yBuffer.remaining();
+        int uSize = uBuffer.remaining();
+        int vSize = vBuffer.remaining();
+
+        byte[] nv21 = new byte[ySize + uSize + vSize];
+        //U and V are swapped
+        yBuffer.get(nv21, 0, ySize);
+        vBuffer.get(nv21, ySize, vSize);
+        uBuffer.get(nv21, ySize + vSize, uSize);
+
+        YuvImage yuvImage = new YuvImage(nv21, ImageFormat.NV21, image.getWidth(), image.getHeight(), null);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        yuvImage.compressToJpeg(new Rect(0, 0, yuvImage.getWidth(), yuvImage.getHeight()), 75, out);
```

Comment:
Why are you converting to jpeg and then to bitmap?  You are doing an uneccesary encode/decode step and probably losing detail in the process.  Can't you just buffer (and use) the yuvimages directly?